packages/flutter/lib/src/widgets/platform_menu_bar.dart

```diff
@@ -383,7 +459,12 @@ class DefaultPlatformMenuDelegate extends PlatformMenuDelegate {
     }
     final MenuItem item = _idMap[id]!;
     if (call.method == _kMenuSelectedCallbackMethod) {
+      assert(item.onSelected == null || item.onSelectedIntent == null,
```

Comment:
Could we assert this earlier closer to the creation point of the MenuItem? Currently, you'd only see the assert if you actually selected the item in the menu

Reply:
Oh, I see below that this is also asserted earlier. Never mind this comment then.

Reply:
Yes, I did both because they're in separate classes. It's probably overkill.